core/blockchain.go

```diff
@@ -2944,3 +3010,14 @@ func (bc *BlockChain) SuperCommitteeForNextEpoch(
 	}
 	return nextCommittee, err
 }
+
+var (
+	leveldbErrSpec         = "leveldb"
+	tooManyOpenFilesErrStr = "Too many open files"
+)
+
+func isUnrecoverableErr(err error) bool {
```

Comment:
how do you define unrecoverable errors? Please add a comment. The conditions is not clear since the reader don't know the scope of all the errors and their full messages.

Reply:
Added more comments. Please tell me whether the comment gives more context about this error. Thanks.

Reply:
thanks. looks good.